presto-hive/src/main/java/com/facebook/presto/hive/BackgroundHiveSplitLoader.java

```diff
@@ -421,6 +430,58 @@ private void loadPartition(HivePartitionMetadata partition)
         fileIterators.addLast(iterator);
     }
 
+    private void addLzoThriftSplitsToQueue(
+            HiveFileIterator hiveFileIterator,
+            String partitionName,
+            Properties schema,
+            List<HivePartitionKey> partitionKeys,
+            ConnectorSession session,
+            TupleDomain<HiveColumnHandle> effectivePredicate,
+            Map<Integer, HiveType> columnCoercions)
+            throws IOException
+    {
+        if (bucketHandle.isPresent()) {
+            throw new PrestoException(StandardErrorCode.NOT_SUPPORTED, "Bucketed table in ThriftGeneralInputFormat is not yet supported");
+        }
+        List<Iterator<HiveSplit>> iteratorList = new ArrayList<>();
+
+        while (hiveFileIterator.hasNext()) {
+            LocatedFileStatus file = hiveFileIterator.next();
+            if (!ThriftGeneralInputFormat.lzoSuffixFilter.accept(file.getPath())) {
```

Comment:
Sure, but what I am doing is mixing the logic from ``LzoInputFormat`` in ``elephant-bird`` and ``LzoIndex`` in ``hadoop-lzo``. I am not quite sure if it's the best way to filter the files by exposing either a static method ``isLzoFile`` or a static member ``lzoSuffixFilter``. We can also return an empty list if the passed file path shows it is not an LZO file.